manual/mapper/config/README.md

```diff
@@ -175,3 +175,82 @@ dependencies {
 
 You'll also need to install a Lombok plugin in your IDE (for IntelliJ IDEA, [this
 one](https://plugins.jetbrains.com/plugin/6317-lombok) is available in the marketplace).
+
+#### Kotlin
+
+[Kotlin](https://kotlinlang.org/) is an alternative language for the JVM. Its compact syntax and
+native support for annotation processing make it a good fit for the mapper.
+
+To set up your project, refer to the Kotlin website:
+
+* Maven: configure [dual compilation][maven_kotlin_java] of Kotlin and Java sources. In addition,
+  you'll need an additional execution of the [kotlin-maven-plugin:kapt][maven_kapt] goal with the
+  mapper processor before compilation:
+
+    ```xml
+    <execution>
+      <id>kapt</id>
+      <goals>
+        <goal>kapt</goal>
+      </goals>
+      <configuration>
+        <sourceDirs>
+          <sourceDir>src/main/kotlin</sourceDir>
+          <sourceDir>src/main/java</sourceDir>
+        </sourceDirs>
+        <annotationProcessorPaths>
+          <annotationProcessorPath>
+            <groupId>com.datastax.oss</groupId>
+            <artifactId>java-driver-mapper-processor</artifactId>
+            <version>${java-driver.version}</version>
+          </annotationProcessorPath>
+        </annotationProcessorPaths>
+      </configuration>
+    </execution>
+    ```
+    
+* Gradle: configure the [kotlin][gradle_kotlin] and [kotlin_kapt][gradle_kapt] plugins in your build
+  script. In addition, declare the dependency to the mapper processor with `kapt` instead of
+  `annotationProcessor`:
+  
+    ```groovy
+    apply plugin: 'kotlin'
+    apply plugin: 'kotlin-kapt'
+    
+    dependencies {
+        kapt group: 'com.datastax.oss', name: 'java-driver-mapper-processor', version: javaDriverVersion
+        ...
+    }
+    ```
+
+You can use Kotlin [data classes] for your entities. Just keep in mind that the mapper expects a
```

Comment:
Neat!  Tried this out (with maven) and it worked!  Thought kotlin may be a problem but cool that it works.  Would be nice if we could use `val`, but hopefully whatever we come up with for [JAVA-2125](https://datastax-oss.atlassian.net/browse/JAVA-2125) solves that.